Plugins/upenn.cbica.captk.brain.survival/src/internal/QmitkCaPTkSurvivalView.cpp

```diff
@@ -0,0 +1,374 @@
+/*===================================================================
+
+The Medical Imaging Interaction Toolkit (MITK)
+
+Copyright (c) German Cancer Research Center,
+Division of Medical and Biological Informatics.
+All rights reserved.
+
+This software is distributed WITHOUT ANY WARRANTY; without
+even the implied warranty of MERCHANTABILITY or FITNESS FOR
+A PARTICULAR PURPOSE.
+
+See LICENSE.txt or http://www.mitk.org for details.
+
+===================================================================*/
+
+#include "QmitkCaPTkSurvivalView.h"
+
+// blueberry
+#include <berryConstants.h>
+#include <berryIWorkbenchPage.h>
+
+// mitk
+#include "mitkApplicationCursor.h"
+#include "mitkStatusBar.h"
+//#include "mitkPlanePositionManager.h"
+#include "mitkPluginActivator.h"
+
+// Qmitk
+//#include "QmitkRenderWindow.h"
+
+// us
+#include <usGetModuleContext.h>
+#include <usModule.h>
+#include <usModuleContext.h>
+#include <usModuleResource.h>
+#include <usModuleResourceStream.h>
+
+// Qt
+#include <QDateTime>
+#include <QFileDialog>
+#include <QInputDialog>
+#include <QMessageBox>
+#include <QFileDialog>
+
+// CaPTk
+#include "CaPTkSurvival.h"
+
+//#include "tinyxml.h"
+
+//#include <itksys/SystemTools.hxx>
+
+//#include <regex>
+
+const std::string QmitkCaPTkSurvivalView::VIEW_ID = "upenn.cbica.captk.views.survival";
+
+QmitkCaPTkSurvivalView::QmitkCaPTkSurvivalView()
+  : m_Parent(nullptr)
+{
+  m_CaPTkSurvival = new captk::CaPTkSurvival(this);
+}
+
+QmitkCaPTkSurvivalView::~QmitkCaPTkSurvivalView()
+{
+
+}
+
+void QmitkCaPTkSurvivalView::CreateQtPartControl(QWidget *parent)
+{
+  // setup the basic GUI of this view
+  m_Parent = parent;
+
+  m_Controls.setupUi(parent);
+
+  //m_CaPTkSurvival->SetProgressBar(m_Controls.progressBar);
+
+  /**** Connect signals & slots ****/
+  // Note: done ahead of time so that initialization can trigger the right events
```

Comment:
> Note: done ahead of time so that initialization can trigger the right events

int val 5; // Initializing integer variable val to 5

xD

Reply:
previously, these were being connected after changes were made to the controls, causing weird behavior -- removing this comment as the code is self-explanatory

Done, will update this PR momentarily